lib/api/core/notifier.js

```diff
@@ -258,8 +258,42 @@ class NotifierController {
    * @param {Array} ids - list of deleted document IDs
    */
   notifyDocumentDelete(request, ids) {
-    return Bluebird.each(ids, (id, callback) => {
-      return this.kuzzle.services.list.internalCache.search(this.cacheKeyPrefix + id)
+    return Bluebird.all(ids.map(id => {
+      const
+        getRequest = new Request({
```

Comment:
The whole point of this work was to send back the body of the document we deleted so we can be notified so I don't really understand why you want me not to fetch the document to get the body.. 
Also at the time we want to notify rooms for the delete notification, no rooms are in the internalCache.